src/main/java/org/folio/edge/api/utils/util/ApiKeyParser.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.edge.api.utils.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Base64;
+import org.folio.edge.api.utils.model.ClientInfo;
+
+public class ApiKeyParser {
+
+  private ApiKeyParser() {
+  }
+
+  public static ClientInfo parseApiKey(String apiKey) throws MalformedApiKeyException {
+    ClientInfo clientInfo;
+
+    try {
+      String decoded = new String(Base64.getUrlDecoder().decode(apiKey.getBytes()));
+      ObjectMapper mapper = new ObjectMapper();
+      clientInfo = mapper.readValue(decoded, ClientInfo.class);
+    } catch (Exception var4) {
+      throw new MalformedApiKeyException("Failed to parse", var4);
+    }
+
+    if (clientInfo.salt != null && !clientInfo.salt.isEmpty()) {
```

Comment:
I think it's better to use StringUtils.isBlank and throw exceptions under if statement, for example:
if(StringUtils.isBlank(clientInfo.salt)){
  throw new MalformedApiKeyException("Null/Empty Salt"
}...